File: math/gcd/gcd.py
"""
Copyright https://github.com/microwind
GCD - Greatest Common Divisor
"""

def gcd(x: int, y: int) -> int:
    while y != 0:
        x, y = y, x % y
    return x

def main():
    print(f"GCD(48, 18) = {gcd(48, 18)}")
    print(f"GCD(56, 98) = {gcd(56, 98)}")

if __name__ == "__main__":
    main()

File: math/gcd/gcd.rs
// Copyright https://github.com/microwind
// GCD - Greatest Common Divisor

fn gcd(mut x: i32, mut y: i32) -> i32 {
    while y != 0 {
        let temp = y;
        y = x % y;
        x = temp;
    }
    x
}

fn main() {
    println!("GCD(48, 18) = {}", gcd(48, 18));
    println!("GCD(56, 98) = {}", gcd(56, 98));
}
